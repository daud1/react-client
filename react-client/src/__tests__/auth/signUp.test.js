import React from 'react';
import { shallow, render, mount } from 'enzyme';
import shallowToJson from 'enzyme-to-json';
import SignUp from '../../components/auth/signUp.component';
import moxios from 'moxios';


describe ('SignUp Component', () => {
    it('should render without crashing', () => {
        const wrapper = shallow(<SignUp />);
        expect(wrapper.exists(<div></div>)).toBe(true)
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<SignUp />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('handles successful axios calls properly', (done) => {
        const wrapper = shallow(<SignUp />);

        wrapper.find('TextField[name="email"]').simulate('change', { target: { value: 'user@email.com' } });
        wrapper.find('TextField[name="password"]').simulate('change', { target: { value: 'password' } });
        wrapper.find('TextField[name="confirm"]').simulate('change', { target: { value: 'password' } });
        wrapper.find('form').simulate('submit', { preventDefault() {} });
        
        moxios.stubRequest('http://localhost:5000/auth/register', {
            status: 201,
            response: {
                'MSG': 'User successfully created!',
            }

        });
        moxios.wait(() => {
            expect(wrapper.html()).toContain('Registered!');
            done();
        });
    });
})