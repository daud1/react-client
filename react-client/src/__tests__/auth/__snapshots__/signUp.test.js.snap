// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Component matches snapshot 1`] = `
<div
  style={
    Object {
      "paddingLeft": "25%",
      "paddingRight": "25%",
    }
  }
>
  <Paper
    circle={false}
    rounded={true}
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": 200,
        "padding": "1%",
        "textAlign": "center",
      }
    }
    transitionEnabled={true}
    zDepth={3}
  >
    <form
      onSubmit={[Function]}
    >
      <div>
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Email"
          fullWidth={false}
          multiLine={false}
          name="email"
          onChange={[Function]}
          rows={1}
          type="email"
          underlineShow={true}
        />
      </div>
      <div>
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Password"
          fullWidth={false}
          multiLine={false}
          name="password"
          rows={1}
          type="password"
          underlineShow={true}
        />
      </div>
      <div>
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Confirm Password"
          fullWidth={false}
          multiLine={false}
          name="confirm"
          rows={1}
          type="password"
          underlineShow={true}
        />
      </div>
      <div>
        <RaisedButton
          disabled={false}
          fullWidth={false}
          label="Sign Up"
          labelPosition="after"
          primary={true}
          secondary={false}
          type="submit"
        />
      </div>
    </form>
    <p>
       Already have an account? 
      <Link
        replace={false}
        to="/login"
      >
        Log In
      </Link>
    </p>
  </Paper>
</div>
`;
